internal/infrastructure/elastic_search/query_device_features.go

```diff
@@ -0,0 +1,141 @@
+package elasticsearch
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+type DeviceFeatures struct {
+	BatteryVoltage struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	FuelPercentRemaining struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Odometer struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Oil struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Soc struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Speed struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Tires struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+}
+
+type DeviceFeaturesResp struct {
+	Aggregations struct {
+		Features struct {
+			SumOtherDocCount int64 `json:"sum_other_doc_count"`
```

Comment:
Are we doing anything with these numbers? Hopefully we've set `size` to be so high that this is always `0`.

Reply:
We have a high enough number,  but I'm including it as a precaution. I intend to check it and log an error if the number is ever above 0

Reply:
That would be really nice! We should probably do that elsewhere.

If it does get too large we can kinda keep limping along with [the composite aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-composite-aggregation.html), which does paginate I think.

There's also [collapsing](https://www.elastic.co/guide/en/elasticsearch/reference/current/collapse-search-results.html), which I haven't looked at yet.

Reply:
Ok.. Since the number is high, i'm thinking of leaving this out in this PR..